dshaur/test-automation
# Web Tests

# Web Automation

* Created 5 test cases for the Sauce Demo Website.
* Implemented custom data provider with .xlsx files. 
* Implemented page objects for the test cases. Implemented a TopMenu AbstractUIObject. 
* Configured properties files.

## Test Cases

1. Login Test:

- Go to the https://www.saucedemo.com/ website.
- Enter valid username and password in the LoginPage.
- Click the login button.
- Verify if an error message was displayed.
- Verify that the InventoryPage is displayed as a result of successful login.

2. Add To Cart Test:

- Perform the login process with valid credentials.
- Navigate to the InventoryPage.
- Add a product to the shopping cart from the list.
- Verify that the product is added successfully to the ShoppingCartPage.

3. Remove From Cart Test:

- Perform the login process with valid credentials.
- Navigate to the InventoryPage.
- Add products to the shopping cart from the list.
- Go to the ShoppingCartPage.
- Remove one of the previously added product from the cart.
- Verify that the product is removed from the ShoppingCartPage.
- Verify that the other product is still present.


4. Cart Checkout Test:

- Perform the login process with valid credentials.
- Navigate to the InventoryPage.
- Add multiple products to the shopping cart.
- Go to the ShoppingCartPage.
- Click on the checkout button.
- Complete the checkout process in the CheckoutPage.
- Verify that the CheckoutCompletePage is displayed as a result of successful checkout.

5. Logout Test:

- Perform the login process with valid credentials.
- Click on the menu icon in the TopMenuPage.
- Choose the Logout option.
- Verify that the LoginPage is displayed as a result of successful logout.
